crates/sui-indexer-alt-framework/src/ingestion/rpc_client.rs

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::ingestion::client::{FetchData, FetchError, FetchResult, IngestionClientTrait};
+use anyhow::anyhow;
+use sui_rpc_api::Client as RpcClient;
+
+#[async_trait::async_trait]
+impl IngestionClientTrait for RpcClient {
+    async fn fetch(&self, checkpoint: u64) -> FetchResult {
+        let data = self.get_full_checkpoint(checkpoint).await.map_err(|e| {
+            if e.message().contains("not found") {
+                FetchError::NotFound
+            } else {
+                FetchError::Transient {
+                    reason: "io_error",
```

Comment:
This reason looks like copypasta from the local client. The local client's errors are `io_error`s because it reads checkpoints from the filesystem, it seems like this should be something else (but not sure what).